_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -504,6 +504,64 @@ public static function register_endpoints() {
 				'permission_callback' => __CLASS__ . '::view_admin_page_permission_check',
 			)
 		);
+
+		// Update settings from the Jetpack wizard.
+		register_rest_route(
+			'jetpack/v4',
+			'/setup/questionnaire',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::update_setup_questionnaire',
+				'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+				'args'                => array(
+					'option_values' => array(
+						'required' => true,
+						'type'     => 'object',
+					),
+				),
+			)
+		);
+
+		// Get settings for the Jetpack wizard.
+		register_rest_route(
+			'jetpack/v4',
+			'/setup/questionnaire',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_setup_questionnaire',
+				'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+			)
+		);
+	}
+
+	/**
+	 * Update the settings selected on the wizard questionnaire
+	 *
+	 * @param WP_REST_Request $request The request.
+	 *
+	 * @return bool true.
+	 */
+	public static function update_setup_questionnaire( $request ) {
+		$option_values = $request['option_values'];
```

Comment:
It may be worth handling the case where `option_values` isn't set.

Reply:
Parameter option_values is defined as a required parameter at the endpoint definition. If it isn't set, the API will return 

> "message": "Missing parameter(s): option_values". 

I don't think I need to/can add any special handling at the update method.

Reply:
Using the value directly without some check makes me feel a bit uneasy as well. But, I don't feel super strongly about it.

Reply:
It may be worth just doing something like the following:

```
$option_values = empty( $request['option_values'] ) ? array() : $request['option_values'];
```